contracts/utils/cryptography/SignatureChecker.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.9.0;
+
+import "./ECDSA.sol";
+import "../Address.sol";
+import "../../interfaces/IERC1271.sol";
+
+library SignatureChecker {
+    function isValidSignature(address signer, bytes32 hash, bytes memory signature) internal view returns (bool) {
```

Comment:
As we discussed offline, ERC1271 signatures that are valid at some point can become invalid in the future, and this is an important difference when compared to cryptographic signatures, which have the non-repudiation property.

I proposed renaming this function to something that highlights this fact. Options:

- `isCurrentlyValidSignature`
- `isValidSignatureNow`

Reply:
I agreed with that comment until I realized that the IERC1271 function is `isValidSignature`. Basically, this is where I got the name from (even though they are actually 2 different functions signature.

I think it would have made sense to rename the IERC1271 function back in the days before it got adopted/implemented by smart wallets. But since this is the underlying ERC name, I think we should match it somehow (`checkIsValidSignature` ?)

Reply:
I don't see why the library name should necessarily be the same as the ERC function name, since it's a utility library and not an implementation of the ERc.